### ISSUE TITLE
Update... would it make sense?

### ISSUE THREAD

**Issue by boelle** [bobc/bobc_hardware#34]
as per subject would it make sense to update Ramps FD ? are there any improvements that could be made?

**Comment by bobc** [comment]
Really, the reason for RAMPS-FD was to help kickstart the 32 bit printer controller era (I was not aware RADDS was being developed around the same time). There are now many 32 bit ARM boards, several using the Due chip (SAM3X8E), so that job seems to be done.
Those boards cover the range from bargain basement to high end feature rich boards, so I'm not sure where an improved RAMPS-FD would fit in. 
There have been a couple of RAMPS-FD forks, but it's unlikely I will do a new version. I mooted the idea a while back, and got a fairly negative reception.

**Comment by boelle** [comment]
Well the problem with RADDS is that that its not open source so you cant check it yourself and it cost an arm+leg as only the people behind them sell it. So i would not count it.... of course its an option and then again my toes are making knots on themselfes

There are of course boards like duet and smoothie, one has onboard stuff and the other has not been updated in a while, and so far firmware like marlin and repetier have not been ported to run with it

I think the negative feedback is due to china took the first versions and just made a pile of them without reading the warnings, as a results there are still a big amount in circulation that only gives the board a bad name.  I still think that ramps-fd has its place but due to the above there is a lot to fight against

btw... when i opened the files in kicad it said that the special lib was missing. is that one you made up or can i download that somewhere?

**Comment by boelle** [comment]
just had a look around and yeah most sellers are still selling rev 1 a and we can thank geeetech for all the negative press :-(

i also looked if any of the forks have been updated and that was not the case

i want to see if i can make improve it, but that missing lib (special) is bugging me. 

if i ignore it there are quite a bit on the schematic that are not connected. not sure if that is related to the missing lib

any chance you have the lib and maybe also a BOM ?

**Comment by bobc** [comment]
It's a no win situation, retain control like RADDS but without FOSS, or go FOSS and lose control to Chinese cloners like RAMPS-FD, Smoothieboard etc. Meanwhile users just buy the cheapest and then complain about lack of support...

From the little feedback I get, there are not many pressing updates to make (apart from add loads of features for zero cost :) ) I guess the two things are the temperature inputs and power supplies. If I had the knowledge, I would add buffers to protect the ADCs, and switched mode power supplies. I think there would also need to be an officially supported LCD control board, that seems pretty standard now.

special.lib is a standard KiCad library that was removed, it is not needed by RAMPS-FD so you can ignore the error. You will need to "rescue" components, because the standard libraries changed. 

The two forks were being sold on OSHPark etc, I had to prod them to publish source files... kinda sad that even FOSS supporters don't follow the GPL! Hard to complain about the Chinese not following the rules if we don't ...

**Comment by boelle** [comment]
oki thanks for the explaination... will try and see if i can manage the rescue after some sleep :-D

But yeah, GPL goes both ways and you cant blame others if you dont comply yourself

Do you have a BOM also ? no rush but it would help if i go ahead and want to make a board for my next printer. sourcing parts  can be a pain :-)

**Comment by rafaljot** [comment]
I have several boards, Duet wifi, RAMPSs 1.4, 1.5, 1.6, MKS gen, sbase, vellman boards,  RAMPS-FD v1 and 2.1
I like the RAMS-FD 2.1 the most. The best versatile shield for Due in my opinion.

What I could want more?
- support for steppers with SPI control, sensorless homing
- better support for 2 x PTC100 instead terminators. and just simple 4.7K for bed.

**Comment by boelle** [comment]
@rafaljot hope you dont mind me asking but where did you get the 2.1 from? so far i have one been able to see version 1

steppers like the TMC2130 etc does not require much, an spi connector is all that is needed and 1 free pin for each stepper driver

if not mistaken PTC needs an preamp, even the duet uses an addon for that. but yes it could be onboard maybe and either enable the addon in firmware or with jumper. it will cost more thou, sadly nothing is free :-)

**Comment by boelle** [comment]
but i guess we can sum up to this so far:

add buffers to protect the ADCs, and switched mode power supplies
if not on board allready an SPI connector
CS pins for the stepper drivers 
either PT100 addon or on board

**Comment by rafaljot** [comment]
Yes, it easy to add SPI but some modern board has more elegant solutions. BTW: Its time to design new stepstick standard....

one guy from Poland has an interesting idea for 12V section. Standard, chip DC-DC step-down mounted over on board. 
https://reprapy.pl/download/file.php?id=8036&mode=view
https://reprapy.pl/download/file.php?id=8081&mode=view

src: https://reprapy.pl/viewtopic.php?f=19&t=2533&hilit=due3dom

**Comment by boelle** [comment]
will have a look but is 12V not a dead horse? TMC2130 runs more silent on 24V. that i have tried myself and yes 24V makes them more silent, i asked watterot myself on that

**Comment by rafaljot** [comment]
RAMPS-FD 2.1 ? ask Yu ( sylyyf2009 at hotmail )

**Comment by boelle** [comment]
"one guy from Poland has an interesting idea for 12V section. Standard, chip DC-DC step-down mounted over on board."

that is what @bobc suggested, but yeah maybe modules are more easy to integrate

yes where did you buy the ramps fd 2.1?

**Comment by rafaljot** [comment]
I answered above
https://reprapy.pl/download/file.php?id=8082&mode=view

look at MS1,MS2,MS3,MS4/SPI configuration

**Comment by bobc** [comment]
Just for info, RAMPS-FD moved to its own repo, https://github.com/bobc/RAMPS-FD/tree/dev
I think I didn't update the BOM because I didn't really finalise the design.

The reason for having 12V and 24V was because some people wanted to run motors at 24V, but also  have fans, LED strips at 12V. There are lots of combinations, it would be a lot simpler to have all 12V or 24V.
I did think about using a separate DC-DC module, but there is no standard.

**Comment by boelle** [comment]
hehehe, @rafaljot game me think link on facebook and i just finished adjust all the 3d stuff. its nice to have some eyecandy :-D

the 2.2 version had far less rescue to do 

a partial BOM would also be usefull

if you dont mind i could make a google sheet with shop links etc. 

how it looks without any mods.... if anything i would change the 3d model for the shield connectors so it  does not include the mega/due board but just the pin headers

![image](https://user-images.githubusercontent.com/3832239/40801124-72379b18-6512-11e8-9318-c0ede1a39660.png)

**Comment by Antscran** [comment]
Hi, Juts been following this thread and this current discussion, so hope you don't mind me adding a comment.  It would be great to see the RAMPS-FD revisited and couldn't agree more with your comment Boelle.
> Well the problem with RADDS is that that its not open source so you cant check it yourself and it cost an arm+leg as only the people behind them sell it.

I have done some design work before and would also help with any prototyping costs if there is enough support for the idea.
Cheers,Ant

**Comment by boelle** [comment]
@Antscran 

so far all i have done is taken the file from @bobc and collected the 3d stuff. My next step would be to try and build up a BOM and then we could varify that the parts are good enough or find better replacements.

I think stuff like the circuits for the bed etc... 

then we could expand it with the ideas that @bobc told

but yeah we need to convise people that Ramps FD is safe and all the negative feedback is LARGELY due to Geetech spamming the market with an version that was not ready

if we can get that far my suggestion would be to create an Organization on github just for the board and maybe a few addons. It might sound like overkill but it allows more than one to have the edit stick

**Comment by boelle** [comment]
Q... how do i edit a WRL file? i want to edit the wrl for the due shield so it only have the pins under the rams fd board... 

i tried freecad but failed

**Comment by bobc** [comment]
Hmm, it seems I have several local changes that I haven't pushed to github. I can probably find time tomorrow to push the latest changes.

I tried getting a quote from Macrofab for 10 off assembled boards, it was around $1200 for 10 boards, which is about 4x the price RAMPS-FD found on ebay. There are some cheaper services based in China, but I didn't get any quotes.

**Comment by boelle** [comment]
i could imagine that getting them made outside china will cost quite a bit more

No rush on the last changes, i think the biggest job if anything would be to convince people that the board has a place.

if you have a partial BOM that would be usefull too

**Comment by boelle** [comment]
i have started the "painfull" process of making a BOM. I decided to split it up in small chunks that follows the schematic

https://docs.google.com/spreadsheets/d/1RLnmdB46gqgr1EFkkU6c9ktHnhKonsIwyvMgGwauXlQ/edit?usp=sharing

for now everyone can comment on the sheet but no edits

should we be so damm lucky that people gets interested and we need to get more organized etc those permissions can be changed :-D

**Comment by rafaljot** [comment]
@bobc ask  this men from China ( @yyf2009,  sylyyf2009 at hotmail com ) as I understand he was interested in producing RAMPS-FD 2.2 and final list price I got was much lower then $1200 per 10.
They talked about it in this thread https://github.com/MarlinFirmware/Marlin/issues/9484
I have two pieces FD2.1 from him, they look better then average China production, definitely better terminals for instance.

**Comment by boelle** [comment]
Ideal it should be us telling the man from china precise what parts we want on the board and what brand each part should have so we know 100% that its the right parts for the job

and there might be small changes to the board also

would say its a bit to early for big production runs.

for that kind of thing we still need to convince many people that Ramps FD is not "junk" etc.

**Comment by yyf2009** [comment]
@boelle @bobc  I already have V2.2 pcb, but I haven't assembled yet. I don't know who needs it. For the V2.1 board, I want to say that I have produced a batch. If you need, you can contact me. Can also produce any board

**Comment by boelle** [comment]
@yyf2009 since V2.2 is alsmost the same as V2.1 we can compare prices... what did the assembled V2.1 cost?

i started up this thread as i think Ramps FD has its place, its just a shame that Geetech took the first versions that where not safe and flooded the market. Everyone things Ramps FD is a BAD board.

We need to do this right and use time to go over every little detail so that everything is checked and rechecked

I have nothing against if the boards are made in china, as long we have  some control over what parts goes on the board. Of course that will also dictate the price

**Comment by boelle** [comment]
and also remember that V2.2 is not 100% complete yet

@bobc still wants some features added to it

- add buffers to protect the ADCs, and switched mode power supplies
- if  not on board allready an SPI connector
- CS  pins for the stepper drivers
- either PT100 addon or on board

no 2 and 3 is for Trinamic Drivers. if its to much trouble we just need to make sure that there are SPI pins there. it could be the SPI connector in the middle that connects to the DUE. We simply just use a stacking header so that we have a socket on the back side and pins on the top side

**Comment by rafaljot** [comment]
I wonder if CFG1-3 jumpers are still needed. 

Is it problem to make additional/optional pads for capacitors on opposite side? Its for better air movement under the drivers. does it make sense? (in K8400 I glued two heat sinks on both sides)

**Comment by boelle** [comment]
we could make them an optional, ie. we dont solder them but we leave the holes in the board

so if end user want to use old steppers they can do that

but i should have a BOM ready by the end of day... or at least a partial one so we can start to fill in the important thing like maker and model of parts

**Comment by rafaljot** [comment]
but if you want to remove jumpers you need to add 3state solderjumpers with configuration SPI/CFG

**Comment by boelle** [comment]
not in this case, 

user just buy the jumpers and solder them on to the board

i guess what you mean with solderjumpers is that we change the holes to pads so user can make a bridge between pads with a blob of solder

but if we change holes to pads the user cant solder in normal jumpers


i think its best to just leave CFG1-3 as they are. and i think @bobc will say i'm correct when he gets back

**Comment by boelle** [comment]
What component supplier woulld be the best?

Mouser? RS Components?

We need one that is world wide and that also sells to private individual. Both Mouser and RS Components do that

**Comment by boelle** [comment]
Benefit with mouser is that we can create a shopping list and share that public so if some people want to solder themselfes we can just give them the board and the link...

Resistors... Do they need to be better than 5%? So far i have just selected 1-5% and picked the cheapest one
LED's.... What current should they be, 20 mA ? So far i have picked 20mA
The EEprom is connected to +V_Logic, do i read correct it can be either 3.3V or 5V depending on Jp101 and if its a due or mega, is that correct?
L1 is a ferrite bead... any data for that?

**Comment by bobc** [comment]
For the BOM, I think https://github.com/bobc/bobc_hardware/blob/master/RAMPS-FD/RAMPS-FD_BOM.ods would be a good start.
Resistor values are not critical, except those in the thermistor circuit. SMD resistors are mostly 1% anyway.
LEDs current is not important.
In theory, RAMPS-FD could run with 5V logic, so to be fully compliant EEPROM should be 3.3V-5V.

Generally, Digikey part numbers are quite universal.

**Comment by boelle** [comment]
Thanks :-D

i did go with mouser and would say i'm 75% through if not more

think i will get the BOM done tonight and would be good to get as many as possible to look at it

**Comment by bobc** [comment]
For the switching 12V power, I was thinking of a module like https://uk.rs-online.com/web/p/switching-regulators/1619073/. It's pin compatible with LM7812, and only requires a little more board space.

I have no problem building my own boards, but I couldn't build boards for other people. So I was looking at assembly houses for making prototypes or small runs for interested people. It turns out I did get a quote from Itead, $957 for 10 boards assembled. I would need to add shipping, import duties and VAT (20%) just to get them to me.

**Comment by boelle** [comment]
i also thought of those modules, i saw a guy use them to replace the voltage reg's in an old C64, worked fine for him and not even an scope could tell much difference

that price is crazy, that is why i'm messing arround with a BOM to see where the price for parts will end, if to high it would impossible to get others interest...

dinner time here, will get the BOM done and then repost link for review. i might have picked stupid expensive parts or the wrong ones

**Comment by Antscran** [comment]
@bobc just curious the price you were quoted was that just for the SMD components on 1 side of the PCB?  Figure costs could be reduced if any through hole components, headers etc were soldered on by the user.

For me buying it in a semi complete kit form like this would be an option, some SMD components would not be an issue but this is not for all.

**Comment by bobc** [comment]
> that price is crazy

Not really, but of course "it depends". For small q's, an SMPS would probably cost me near £2 just in parts, so actually a module like that is much simpler and smaller, unless I am making 1000 quantity. But then if I was in Shenzen, I could make it for maybe 50p and still sell on ebay for £1 incl shipping.

>  if to high it would impossible to get others interest...

That is #1 problem I face (and Smoothie, RepRapPro etc). I can make it good, or cheap , but not both. People always prefer cheap. In fact they will buy the cheap board from China then expect I give them free lifetime support because I designed it :) I've kinda moved to the position where if people want cheap, I don't care about them because I could only do that if I lived in Shenzen.

**Comment by bobc** [comment]
>  just curious the price you were quoted was that just for the SMD components on 1 side of the PCB? Figure costs could be reduced if any through hole components, headers etc were soldered on by the user.

All components. Unfortunately there is an infinite range of users requests from part kit to built and tested. Horrible logistics. Kits would actually cost more because of the time to kit out parts.

Oh yeah, I forgot to add cost of testing to sale price :)

**Comment by bobc** [comment]
So what type of Trinamic modules are people using? There are these on ebay https://www.ebay.co.uk/itm/TMC-2130-2100-2208-Stepstick-Stepper-Motor-Driver-Module-Heat-Sink-fr-3D-Printer/282760855753?hash=item41d5dc40c9:m:mYc2X5y_VTtEvKPe3Jxu4cQ, originally designed by Waterott I think.
It looks like it would be quite easy to support those with just a few extra resistors and still be compatible with classic Pololus.

**Comment by bobc** [comment]
> Is it problem to make additional/optional pads for capacitors on opposite side?

It's not easy, because the routing of power tracks is quite cramped.

A possibility is to increase board size, that might be needed anyway to allow for new features and improve the layout. It is already a struggle to route on 2 layers. 4 layers would make it easier but more $$$.

**Comment by bobc** [comment]
I've modified the stepper circuit to allow for Trinamic modules, it adds 2 resistors

![image](https://user-images.githubusercontent.com/972103/40856170-d55c168a-65ce-11e8-994a-ed9e72395939.png)

However, to make it work, I also need to add 3 jumpers to optionally connect SPI bus to all the modules, or add 3 resistors to each stepper module. Otherwise, adding a microstep jumper will stop the SPI working...

**Comment by boelle** [comment]
> So what type of Trinamic modules are people using

I think its mostly 2130 and 2208, the latter can handle more current and have a new version of stealthchop

**Comment by boelle** [comment]
> However, to make it work, I also need to add 3 jumpers to optionally connect SPI bus to all the modules,

and each modules also would need a CS line, it would require a person that is good at routing tracks :-)

**Comment by boelle** [comment]
>  In fact they will buy the cheap board from China then expect I give them free lifetime support because I designed it

hmm.... that is not fair. as i understand it you and the people working on it at the time did put a very clear warning that is was not ready yet.

**Comment by boelle** [comment]
> Kits would actually cost more because of the time to kit out parts.

We could get arround that by creating a "project" at mouser and share the link. That would enable people to order all the parts on their own. And we could even do the same at oshpark so people can order a board. just thoughts thoug

**Comment by bobc** [comment]
On second thoughts, I need to use jumpers for each SPI line, and 1 for each CS, so another 9 jumpers. Otherwise the SPI will mess with the config of regular Pololus.

The 2130 has SPI, the 2208 has a unidirectional UART. The UART can also be accommodated I think (with another jumper!).  The fysetc version is slightly different to the Watterott version.

> We could get arround that by creating a "project" at mouser and share the link. 

Sure, good idea.

**Comment by boelle** [comment]
Made it to the end of the BOM and have used the one you linked to where i was in doubt

Q's:

- is C4 the same as C3? i assume yes but assumptions can sometimes be fatal :-D
- U4 the 5V regulator is now mentioned anywhere, how many amps should it supply?
- L1 is not mentioned anywhere, from the kicad files i assume its an ferrite bead, will any that fits the footprint do the trick?
- C301 should be a c_elec_4x5.3 according to the BOM, but the board say 0603, voltage is not mentioned, should it be a 16v ?
- c1001 is not mentioned in the BOM but from the board i found that its a c_elec_4x5.3, should it also be a 16V?
- R211 is marked as NF, does the mean not fitted?
- C601 is a 0603 on the board, is that also a 16V?
- D102 BOM says diode_do41 and 1N4004, but schematic says SS10Pxx, but i'm just confused

Now rechecking the BOM

- P101 is only rated at 12A so we need to find an replacement that can do at least 16A
- Not sure if it makes sense cost wise to use different rating for P107 and P108 
- The fuse holders are spec'ed at 15A, with the bed at ~16A is that not a tad to the low side?

**Comment by bobc** [comment]
After a bit of digging, I found the BOM I create for Itead quote on v2.2. https://github.com/bobc/RAMPS-FD/blob/dev/RAMPS-FD/RAMPS-FD_bom_Itead.xlsx

>    is C4 the same as C3? i assume yes but assumptions can sometimes be fatal :-D

yes

>    U4 the 5V regulator is now mentioned anywhere, how many amps should it supply?

I specced a 1.5A part, but 1A should be ok.

>    L1 is not mentioned anywhere, from the kicad files i assume its an ferrite bead, will any that fits the footprint do the trick?

I have https://www.digikey.co.uk/products/en?keywords=445-2166-1-ND 

>    C301 should be a c_elec_4x5.3 according to the BOM, but the board say 0603, voltage is not mentioned, should it be a 16v ?

I changed C301..C304 to 0603 ceramic. I couldn't see any reason for them to be electrolytics.

>   c1001 is not mentioned in the BOM but from the board i found that its a c_elec_4x5.3, should it also be a 16V?

Can be, but it's part of the faulty thermistor filtering which I recommend not fitting.

>    R211 is marked as NF, does the mean not fitted?

Yup :)

>   C601 is a 0603 on the board, is that also a 16V?

For simplicity I specced all ceramic to be 25V, although I think 16V is Ok for most of them. The price difference is minor.

**Comment by boelle** [comment]
My version of the BOM with only parts from mouser and only just the numbers that is needed for one board comes out to 65€

The screw terminals is not included as they where only rated for 12A and the bed do 16A
Fuse holders are spec'ed at 15A so might just do the trick

**Comment by boelle** [comment]
The BOM: 

https://docs.google.com/spreadsheets/d/1RLnmdB46gqgr1EFkkU6c9ktHnhKonsIwyvMgGwauXlQ/edit?usp=sharing

hope it makes sense

will recheck to see if i screwed up on a footprint etc, what could be interesting is if we can get it cheaper and still only buy just what is needed for one board

For the fun out it i uploaded the board files to oshpark and 3 boards cost 77$ thats roughly 22€ so its 87€ for one board without any assembly done

**Comment by bobc** [comment]
I got €328 for complete BOM quantity 10 at mouser(uk) ... there is a decent price break at x10 but that is quite a difference. 

The EEPROM should be < €1, e.g. https://www.mouser.co.uk/ProductDetail/511-M24512-RMN6TP

OSHPark is shockingly expensive compared to Chinese sources. I usually get boards from Itead.

**Comment by boelle** [comment]
hehe, i agree on oshpark, but they where the only i knew could take the kicad pcb file directly like they do with eagle files 

Do you think we can get the parts much cheaper if only going for 1 board at a time? and just the usual crazy thought, how many boards do you think would be needed in a batch before it can compete with the  unsafe version allready on sale?

EDIT: the cheaper EEPROM knocked the price for parts down to 60€

**Comment by bobc** [comment]
> Do you think we can get the parts much cheaper if only going for 1 board at a time?

For things like pin strips/header I buy them in bulk off ebay then cut to size. That's ok for DIY, but not for volume manufacture. I also buy smaller connectors off ebay, the quality is good enough. For the main high current power connectors I get from RS, Digikey etc. For resistors,caps,LEDs etc I buy 50 or 100 but I use them in other projects.

There are basically two ways to reduce the unit cost; build a LOT of units or build it in China. At about x1000 the volume discount reduces the BOM cost to about 50%. Manufacturers like Geetech will be buying full reels from local market at wholesale prices, plus they have cheap labor costs. The company I worked for prices parts in x10,000 quantity, and they weren't really a high volume manufacturer. They get parts from distributors like Arrow or direct from manufacturer.
To get near to ebay/aliexpress RAMPS-FD prices is basically impossible if manufactured in small quantity, or in the West. I guess a company like Olimex could make it for about 2x ebay price.

**Comment by bobc** [comment]
I changed the fuse holders to https://www.mouser.co.uk/ProductDetail/Keystone-Electronics/3568?qs=%2fha2pyFaduhjKj2TpGWSG6EII8c9zrvT6dr%2fadQGxMA%3d, they are slightly cheaper than the Littelfuse.

**Comment by boelle** [comment]
and most important they are better rated, i was a bit worried that the 12A was not enough for things like the bed

EDIT: the fuse holders knocked another 5€ off
EDIT2: i just downloaded the 3 hour old copy of the board and i can see that also the reset switch is smd now. will change that tomorrow

and for some odd reason not very much of the 3d shapes show up and some are not rotated right, i guess that the eyecandy side of things would come last

**Comment by bobc** [comment]
I looked into the current ratings quite a lot. Current ratings are a little fuzzy. e.g. for the fuse holder it has 3 ratings, UL;20A, UR;15A, CSA:10A. They use different test scenarios and come up with different ratings. Those will have some margin built in. Current ratings are thermal ratings, if the component has free movement of air around it will have a higher effective rating than if packed next to others in a closed box.
The advantage of branded parts is that you can be sure the part really is made to that rating, so provided you stay within 20% or so it should be fine. OTOH, the chinese knockoffs will rate parts at 15A but they are only good for half that.
Note though that Molex etc do the same type of connector with different ratings (due to the type of plastic used). It is tempting to pick a cheaper version but it might have a lower rating.

For those type of connectors, 15A is around the max, to go higher would need a larger connector. 16A seems high for heat bed, thats nearly 200W. At that point it's a whole lot better to use 24V.

**Comment by yyf2009** [comment]
@boelle  FD V2.1 about  $21/pcs

**Comment by boelle** [comment]
@bobc about the amps, it was me that messed up a bit, the mk42 heat bed is only 12V but ~0.8 ohms so its ~180W, i just remembered it as more than that. Not sure if it would be ok to drive the same bed with 24V

strange enough orballoprinting.com also do a clone of the mk52 bed ( the one with magnetic steel sheets ) and that one should be a 24v one, but their picture say 12V, not sure why

well time for me to get my printer running... getting close to where i want things 

final thought, would a V2.2 at arrund the 22 $ mark be more a competitor ? of course we would still need to retain control of the parts used etc

**Comment by boelle** [comment]
@yyf2009 would it make sense for you to sell the V2.2 instead once the design is finalised? or do you have to many boards in stock

**Comment by rafaljot** [comment]
Does it make sense to reconnect power source for heaters and join with the bed ?
It is easy tu buy 24V heat bed, 24V 30,40W cartridges. Good fans are 12V only (and 5V), like Sunon Maglev MF50151VX-A99, Noctua NF-A4x20. 

Using 24V Power Supply + Step-down 12V 2-3Amp I can make this configuration.
Or 24V bed and heaters, 12 step-down fans+etc, 30+ step-up for motors. etc.

**Comment by yyf2009** [comment]
@boelle    There are 10 pcb v2.2, I have not yet started to assemble, if you complete the design, I can produce, and sell them

**Comment by boelle** [comment]
@yyf2009 please dont start to assemble them. @bobc have not finished the pcb design

**Comment by yyf2009** [comment]
@boelle Do I need to wait for a new design file? , how long it will take?

**Comment by boelle** [comment]
@yyf2009 i dont know how long it will take for @bobc to finish and release V2.2

if you look here: https://github.com/bobc/RAMPS-FD/tree/dev

it clearly says that nothing is ready for release yet, so the 10 boards you have is a waste

where did you get your v2.2 design files from?

**Comment by yyf2009** [comment]
@boelle from bobc

**Comment by boelle** [comment]
but @bobc has not relased the files yet

do you have a link to where you got them?

**Comment by yyf2009** [comment]
https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD

**Comment by boelle** [comment]
@yyf2009 

you have not read this file: https://github.com/bobc/RAMPS-FD/blob/dev/README.md

**Comment by bobc** [comment]
Sorry, I forgot to update the readme when I last worked on the dev version.

I don't have any plan to update 2.2 in the foreseeable future, so the current dev version is as good as any. Of course, now that people are building "2.2" even though it has not been released or even finished (deja vu), I would have to call the next iteration 2.3...

**Comment by boelle** [comment]
@bobc i just told our friend from china that updates could happen and that just because the files are there it does not mean they are good for production

ie i wanted to prevent that we have a repeat of what a other well know china company did with the first unsafe version

**Comment by bobc** [comment]
What do people think of https://reprap.org/wiki/RD3D/1.0 ?

**Comment by boelle** [comment]
looking, it smells a bit like the source is your board

**Comment by bobc** [comment]
It's based on RAMPS 1.4.2 which was done by German RepRap. It probably works out cheaper than RAMPS-FD.

**Comment by boelle** [comment]
the thing i'm concerned about is if the board layout is done right when it comes to current etc. and that analog stuff is isolated from the digital so noise does not go from one side to the other

and then if the prober parts are used

**Comment by bobc** [comment]
There is also https://github.com/dvjcodec/Due3Dom, which is the Polish board mentioned earlier, but it is not Open Source. But there might be some ideas there, the layout is much cleaner but the board is quite big...

**Comment by tig33r** [comment]
There is also DUE3DOM mini and DUE3DOM LE. As far as I know all three boards has source on github.

**Comment by bobc** [comment]
"Has source" is not the same as "Open Source"! It's only Open Source if it has an open source license, and CC-BY-NC is not Open Source.

I didn't find a DUE3DOM LE on github.

**Comment by boelle** [comment]
@bobc what is your own stand on the RD3D board?

**Comment by bobc** [comment]
RD3D has a few features more than RAMPS-FD, e.g. onboard SD and support for SPI steppers, otherwise it seems similar. The designer seems to have ignored RAMPS-FD, not sure why. The board was costed at $15/100 (Chinese production), but little activity for a while.

**Comment by boelle** [comment]
hmm... i could live without the SD and SPI, but also i'm not scared soldering wires on the adapter for the display etc

but if you at some point add that to v2.2 or 2.3 i would go for that and assemble a board myself

with all the work and experience that have been put in to Ramps FD it would be a shame to throw it out  the window

**Comment by boelle** [comment]
Just a thought that hit me last night...

i know there might be a price difference, but would the Teensy 3.6 be a better board than the due?

**Comment by boelle** [comment]
have played a bit with the thought on the schematics... since Teensy is 3.3v only a few things can be removed.

I started out with the power schematic, does this look about right?

![image](https://user-images.githubusercontent.com/3832239/41232611-3252a268-6d87-11e8-8f9e-93895e3df10b.png)

**Comment by boelle** [comment]
I wonder for the VCC rail, what is the draw on that? the Teensy have 250 mA on its 3.3V output
if the draw is below that we could simply make the VCC rail use that

EDIT: i looked at the emergency stop sheet where the buffers are and read datasheets on them. If not mistaken they need 5V to run, but i cant see where power is connected to them? 

EDIT2: ahh, i clicked show hidden pins and they came up.

check my logic here for the emergency stop sheet

since the teensy is not 5v tolerant rp2 and rp3 needs to be connected to 3.3V but the buffers themselfes needs to be connected to 5v. or is there no need to change anything?

**Comment by boelle** [comment]
@bobc would it be easy for you to adjust the design to fit on the Teensy? i have given it some more thought and i think it could be an great option.

Sadly i lack the skills to do this 100% on my own

**Comment by bobc** [comment]
I wouldn't favour Teensy as they are not Open Source. Anyway, they don't really have enough pins.

**Comment by boelle** [comment]
oh well.... the lack of pins makes the difference

do you have any work not pushed to github? i'm going to finish my hybercube next month and want to use Ramps FD and need to plan a bit ahead so i can order parts etc

**Comment by boelle** [comment]
@bobc 

did you make the changes to allow for tmc2130? if so are they pushed to github so we all can have a look?

i will spend the day going over the BOM i made up and start prep work to order parts. I will order enough parts for at least 3 boards so i have spares, but my plan is to make 2 boards so i can fit them on both my machines

**Comment by bobc** [comment]
> do you have any work not pushed to github?

I started a change to make the aux connectors plug directly to an LCD panel instead of needing an adapter board, but I don't have time to finish it.

> did you make the changes to allow for tmc2130? if so are they pushed to github so we all can have a look?

I updated the schematic to add some jumpers and pull up resistors on SLEEP/RESET. I added the resistors to the pcb and routing the traces. Those changes are in github, 

BUT I did not place the jumpers, it would take a complete re-routing to fit them in, probably need to increase board size. Since then I noticed a smarter way to do the jumpers from another project, but again I don't have time to do any of that.

It seems in some places TMC2130 are not regarded highly, I have never used them so couldn't give an opinion.

**Comment by boelle** [comment]
The problem with 2130 is that they run more hot. And  they do require 24V to be silent. Some people dont know that they can feed them 24V by removing the Vmot pin and connect it to an 24V source

and then of course there is the lot of misinformation and china knockoff's that dont help it either.

If it was me i would go for the TMC2208 instead, it runs cooler and can handle more current, but it requires 2 serial lines. But maybe 2 serial lines are more easy to do than SPI.

So what you have pushed so far on https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD seems to be ok for me, i dont mind running a few "bodge" wires to make it work as it can be done quite nicely if i take my time

**Comment by craftoscar** [comment]
@boelle 
@bobc 
so to this day is the branch of main (v2.1/V2.2?) or dev (v2.2/2.3?) are ready to use?  
I would like to find a pre-build RAMPS-FD v2.1 form @yyf2009, but the shipping cost is too irrational.
so I'm trying to find other alternatives, it's good to see you guys are here thu.

BTY:
A page in [Reprap](https://reprap.org/wiki/RAMPS_1.7) shows a retired man call MrAlvin are beginning his work to design RAMPS 1.7
where is the post in discussion https://reprap.org/forum/read.php?219,824661

they seem to have some interesting idea about jumper's layout to support TMC2130s

**Comment by boelle** [comment]
@craftoscar 

no @bobc has not released v 2.2 as ready for production, not sure about v 2.1

**Comment by craftoscar** [comment]
I am shure @yyf2009 had some v2.1 ver.B in his hand. and I welling to purchese it excpt for the shipping fees
I notice that a guy in [Poland](https://www.elektroda.pl/rtvforum/topic3455800.html#17184419) are selling? v2.2 PCBs

by pic it caps 6 MOS so that probbly had onbord 5V MOS intergated.
like @bobc mentioned in post years ago to his vision in Ver 2.3

by the soure in [https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD](https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD) 
I visit in June 18 here (GMT+8) in Taiwan



![capture0000](https://user-images.githubusercontent.com/40343420/41512951-35256eaa-72c6-11e8-8d37-578d97a18195.PNG)

![capture0001](https://user-images.githubusercontent.com/40343420/41512952-3b038618-72c6-11e8-9870-13cdb93b2058.PNG)

shows the mark on PCB is v2.3 now it share the simmlar layout with the Polish site

For alternative I like to make one for myself thats a bit confusing 

I need to know witch one are clear to go 

if any one want to get together to mess produce the PCBs I would like to join! 


the reson why took me here is that I just fried my MKS Gen L in last thermocouple test 
My intention is to use some Analogue instrument AMP (like AD595/597/8495) as Temp sensor to print Poly Carbonate.
but I had some hard time to find the geinume schemeatics form MKS to tinkiering it
And the expriance to hock their MKS TFT32 touch screen just suck.

So I start to find open source boards and RepRap page lead me here.

**Comment by boelle** [comment]
non of them are clear to go.... as said before they are not ready for production.

**Comment by craftoscar** [comment]
I got it, I thought that @bobc are still fine tuning V2.3 now
I am really looking foward to this borad, since it already cumalated lot of improvemnts.
I think this borad deserve more popularty, and I would like to made some for my frinds new in 3D printing, or possibly my 2nd printer in future .

To me the V2.1 s are good to go,In conditioning for instrumental AMP for thermocouples
(yeah they'er very sensitive to voltage referance)

**Comment by boelle** [comment]
yep he is doing that, but it will take a long time.

I'm also looking for new electronics  for my 2 printers and i'm very tempted to use v2.2 from https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD

both v2.1 and 2.2 are just versions. I think @bobc increase versions when a step is made. It does not mean its good to go, so we are making board at our own risk.

**Comment by boelle** [comment]
@craftoscar 

btw.... if you have not read it, the reason why Ramps FD have a bad name stuck to it is because some factories took the very first versions and mass produced it. A lot of people got boards that was not safe and people risked burning down their house.

and a lot has happended since.

to give Ramps FD a good name again it must be updated and tested. and if it should be able to to compete it must be same price as what you can get on ebay

but to make that happen we need a factory we can trust and they must use only good parts that we can give green light on

but again @bobc has a lot of other things to do and i think Ramps FD is not a big thing for him

So we will need a lot of patience, if i remember correct he is trying to add support for TMC2130 and that requires a lot of work. And there are a few other things i forgot that he is also working on

**Comment by craftoscar** [comment]
@boelle 
> btw.... if you have not read it, the reason why Ramps FD have a bad name stuck to it is because some factories took the very first versions and mass produced it.

I knew, I had read thoes post pepole use Geeetech 3D based v1s...

> to give Ramps FD a good name again it must be updated and tested. and if it should be able to to compete it must be same price as what you can get on ebay

but to make that happen we need a factory we can trust and they must use only good parts that we can give green light on

but again @bobc has a lot of other things to do and i think Ramps FD is not a big thing for him

Yes, there sure have some fab are will cabable and good QC bulid in
As a Taiwanese live aside China, I know OEM are the main foruse in this region
but things goes to OEM qunatity DO maters.
We Taiwanese likes to hosts small production (maybe 100~ frew k pce.) for some costomize gadgets 
looks like kickstarter, but mostly designer already had good working prototypes aside
their intention is to make their product commercialize, ro perhase make a new company themselfs

witch is quite diffent aspect then 3D printer electronics here we are

**Comment by craftoscar** [comment]
And to 3D printer electronics, Dozen of company based in China already develop their own 
feature rich 32 bit contrl boards in very resonable price
(even the borads like Maga+RAMPS1.4 or Due+RAMPS-fd can't compete with.)
But problem is that the support are not good enought for costomer ouside China
And documentation are insufissent for us to tinkering or Mod around it.
To make mod working u may need to buy their upgrad moudels (witch normally 1/3 about their MB prise).
I not bleam them about the price, this are their working business mode.
but to me some of their moudels are really over priced and you can't really find how they worked when it fail.

To make me simple I just want to find prize reasonable, simple ,easy to fix electronic to work with

**Comment by craftoscar** [comment]
> So we will need a lot of patience, if i remember correct he is trying to add support for TMC2130 and that requires a lot of work. And there are a few other things i forgot that he is also working on

To TMC2130 I not very look forward to it too much (at least for high performance or big volume printers)
the low current output and thermal problem keep me away from it.
I will be held to see if Trinamic will change their interface to another one like they done in TMC2208
but AFAIK their possible next-gen stepstick (TMC-5160 according to a post on the twitt) probably will stick on SPI

**Comment by boelle** [comment]
problem with the drivers are that a lot of misinformation flows on the internet

i have so far only seen one on youtube that did more serious testing where he got the temp of 2130 and 2208 and compared them. and because 2208 is proven to run cooler than 2130 and they can handle a little more amps that is why im more interested in 2208

**Comment by craftoscar** [comment]
> 
problem with the drivers are that a lot of misinformation flows on the internet

https://youtu.be/GVs2d-TOims?t=3m17s
is this whom you are talking about?

if yes, then he is the person who gives very clear and in detail explain to the facts behind the specs.
which is depends on RDS of those chip, and this is just another reason why I still stick on DVR 8825s

it has RDS range form 0.2 to 0.32 Ω compared to Trinamiscs are 0.4 to 0.6 and 0.28 to 0.39 for TMC 2130 and TMC2208s respectly.

those number implies that TI's drivers are more power efficient when doing it's jobs!

**Comment by boelle** [comment]
i only buy TMC drivers from Watterot in germany and no cheap china build ones :-D

but yes that youtube guy is one of a few that does a great job explaining what nobody wants to hear: you need to do your researh before you do anything :-P

**Comment by boelle** [comment]
@bobc 

am i correct that https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD has what is needed except wires for 2130/2208 ?

**Comment by bobc** [comment]
Yes, I think so. Bear in mind that version is completely untested, use at own risk etc.

**Comment by boelle** [comment]
yep, there is nothing like being the guinea pig :-D

My plan is to get a stack of boards made and slowly assemble one testing as i get along

But not sure when that will be

**Comment by boelle** [comment]
@bobc 

So i got the projects list cleared and will now get this one ready so i can order a few boards and start to assemble one

i know it does not matter right now but on https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD  some 3d shapes are missing

a few of them are:

rmc_packages3d/Keystone/keystone-PN3568-k.wrl
rmc_packages3d/misc/fuse-mini-bussman-20A-k.wrl
walter/conn_screw/mstba_2,5-4-g-5,08.wrl

are those published on your other repros or are they taken from somewhere else?

**Comment by boelle** [comment]
oh, and sorry to bother again

do you have any part numbers you can recommend for P101, P107 and P108 (biggest screw terminals)

those 3 are about the only parts i have not found so far

**Comment by boelle** [comment]
@bobc 

found part numbers for those screw terminals so no worries there

but i looked at P801 and P802 and since they are for endstops i wonder why they are not next to each other so they form a 3 way connector for each endstop

is it so that the connector can be keyed to prevent it going in the wrong way?

**Comment by craftoscar** [comment]
@bobc 
Would it possible to add this [https://shop.pimoroni.com/products/adafruit-precision-lm4040-voltage-reference-breakout-2-048v-and-4-096v](url) LM4040 precision voltage reference along with RAMPS-FD?
I find some problem for my AD8495 thermocouple AMP with DRV8825 stepper drivers.
My config is hoock AD8495 on to RAMPS-FD 'S A4 PIN and supply it with nearby 3.3v around AUX- 1region, I got some temperature spikes after moving my printer's carriage in x+ direction
 
I do some home works, found out it probably related to this  
[https://reprap.org/forum/read.php?13,22502,22521#msg-22521](url)

I think that would probably tribute by DRV8825 pulls down some ref. voltage form DUE, even thought I already supply my stepper drivers with a separate PSU

**Comment by 4Joe** [comment]
I have made this pcb through some company in China for personal interest. Does anyone would like to share the update BOM of this board?

**Comment by boelle** [comment]
i have made a BOM for v2.3: https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD

it has NOT been checked by me or anybody else

https://docs.google.com/spreadsheets/d/1RLnmdB46gqgr1EFkkU6c9ktHnhKonsIwyvMgGwauXlQ/edit?usp=sharing

**Comment by yyf2009** [comment]
@bobc  In the V2.3 version, the temperature control circuit has been changed. Does this have any effect on the actual measured temperature?

**Comment by bruce356** [comment]
@yyf2009 hi, what is the delivery time for the "Arduino due ramps_fd v2.1 expansion board " do you have stock or does the board need to be assembled.
thanks - bruce

**Comment by yyf2009** [comment]
I shipped it on August 27th, Beijing time.

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

________________________________
发件人: bruce <notifications@github.com>
发送时间: Wednesday, August 29, 2018 9:03:33 AM
收件人: bobc/bobc_hardware
抄送: Yongfeng Yu; Mention
主题: Re: [bobc/bobc_hardware] Update... would it make sense? (#34)


@yyf2009<https://github.com/yyf2009> hi, what is the delivery time for the "Arduino due ramps_fd v2.1 expansion board " do you have stock or does the board need to be assembled.
thanks - bruce

―

**Comment by yyf2009** [comment]
I want to produce a ramps_fd v2.3 board, but I don't know if this temperature will be accurate? After all, this schematic has changed.

**Comment by bruce356** [comment]
@yyf2009 , thanks,  lets hope @bobc gets back to you regarding the temperature control circuit.
regards - bruce

**Comment by boelle** [comment]
@bruce356  & @yyf2009 

it would make more sense to start an issue here: https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD

that is where development happens

if @bobc dont want to get to much involved anymore i can understand him.

he and another put much time in to it and some well know company from china did a bad move flodding the market with the first version that was not safe. Even thou this well know company got many warnings they spammed the market with unsafe boards and gave Ramps FD a very bad name. So a lot of work and only a bad rep left. I myself would have given up

since there are only very "expensive" 32 bit boards out there i still think Ramps FD can have its place

What we need is people with knowledge of schematics and electronics design. My own skills are VERY limited. But i have kicked some Kicad skills in to myself so i can build a schematic in kicad and i can do simple board work (Remeber SIMPLE)

What i have on my desktop is the board as it is on https://github.com/bobc/RAMPS-FD/tree/dev/RAMPS-FD

I fixed a lot of 3d shape stuff, it does not change anything but hey why not have some eye candy if it not that much work

**Comment by bobc** [comment]
> Does this have any effect on the actual measured temperature?

I removed the over-voltage protection, so it should be more accurate.

As I have said numerous times before: 

1 - I don't have time, facilities or inclination to do further development work on RAMPS-FD.

2 - The schematics and PCB layout in the dev branch are UNTESTED and USE AT YOUR OWN RISK. If you don't understand why there are changes in the schematic, you should not be using them!

The designs are provided for the interest of anyone who wishes to use them within the terms of the GPL.

**Comment by boelle** [comment]
that is also why i kind of suggested that if Ramps FD should move forward we need more people on board that has the skills that it takes. i for sure does not have all the skills needed, some i have but not all
 
and not to be rude or anything but @yyf2009 does not have the skills as he asks questions about the schematic. but he might be a top PCB maker.

on that regard i can only compare the thermistor part of the schematic to what i know that works and that is the Ramps 1.4 board. since they to me look the same that is not where i'm worried

and the stepper driver part also looks very familiar to ramps

where i would start to sweat and get nervous is adding onboard support for watterott's TMC2130 and TMC2208, i could start the work by my confidence level is not high enough and i need someone else to check my work. 

its not a simple task as both SPI needs to be routed to each driver and also individual serial lines

**Comment by yyf2009** [comment]
Hi Bruno，
   I ship on August 27th and arrive at your delivery location on August 29th.，I think you should ask your container warehouse.



[cid:image001.png@01D45E12.5032B630]

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

________________________________
发件人: bruce <notifications@github.com>
发送时间: Saturday, October 6, 2018 6:21:58 PM
收件人: bobc/bobc_hardware
抄送: Yongfeng Yu; Mention
主题: Re: [bobc/bobc_hardware] Update... would it make sense? (#34)

Hi Yongfeng Yu, the Ramps FD v2.1 and LCD Adaptor have not arrived yet. Any idea from your end as to where it could be.

Thanks and regards C Bruno

2018-08-25 Order No .: 176248242606646697

**Comment by bruce356** [comment]
Hi Yongfeng, thanks for the reply, I do not understand what you have done or where you sent the card?

*         2018-Aug-27Mon.16:05:25[Jiaxing City] Hailing's Bai Lingbin [18806833302] has been collected

*         19:25:24[Jiaxing City] Express mail has been issued from Haiyan, ready to be sent to Jiaxing Transit Department

*         21:31:50[Jiaxing City] Express has arrived at Jiaxing Transit Department

*         21:35:31[Jiaxing City] Express has been sent from Jiaxing Transit Department and is ready to be sent to Guangzhou Center

*         2018-Aug-28Tues.21:03:14[Guangzhou] Express has arrived in Guangzhou Center

*         21:24:49[Guangzhou] Express has been sent from the Guangzhou Center, ready to be sent to the center of Dongguan

*         2018-Aug-29Wed.00:26:08[Dongguan] express mail has arrived in Dongguan center

*         01:30:30[Dongguan] Express has been issued from Dongguan Center, ready to send to Dongguan Shatian

*         04:14:14[Dongguan] Express has arrived in Shatian, Dongguan

*         10:33:37[Dongguan] Dongguan Shatian dispatcher Xiao Baichao 2 18027574003 is sending you a piece

*         15:00:28[Mainland] Your parcel has arrived at the consignment bin.

*         15:01:44[Mainland] Your parcel collection bin has been signed

*         16:37:42[Mainland] Your parcel has been put into storage. If the logistics information has not been updated for a long time, it is recommended to consult the courier company.

 

“Your parcel has been put into storage” what does this mean? I am in AUSTRALIA not in CHINA, please clarify this for me 

Thanks and regards – Bruno

**Comment by yyf2009** [comment]
Please enter [My Taobao], find the [Official Collection] button on the left side of the page, click [Together Order], click on your transporter icon.

1. If the goods have already displayed “Waiting for Buyers' Orders”, check the item order to make a combined payment~

2. Check the items to be shipped. You can check the weight of your order and the total order weight, total shipping cost, check the weight and shipping cost and click the button of [Freight Settlement].

3. After the order is merged, confirm the receipt method and the delivery address to ensure that your goods can be successfully cleared and delivered.



发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

发件人: bruce<mailto:notifications@github.com>
发送时间: 2018年10月7日 19:14
收件人: bobc/bobc_hardware<mailto:bobc_hardware@noreply.github.com>
抄送: Yongfeng Yu<mailto:sylyyf2009@hotmail.com>; Mention<mailto:mention@noreply.github.com>
主题: Re: [bobc/bobc_hardware] Update... would it make sense? (#34)

Hi Yongfeng, thanks for the reply, I do not understand what you have done or where you sent the card?

* 2018-Aug-27Mon.16:05:25[Jiaxing City] Hailing's Bai Lingbin [18806833302] has been collected

* 19:25:24[Jiaxing City] Express mail has been issued from Haiyan, ready to be sent to Jiaxing Transit Department

* 21:31:50[Jiaxing City] Express has arrived at Jiaxing Transit Department

* 21:35:31[Jiaxing City] Express has been sent from Jiaxing Transit Department and is ready to be sent to Guangzhou Center

* 2018-Aug-28Tues.21:03:14[Guangzhou] Express has arrived in Guangzhou Center

* 21:24:49[Guangzhou] Express has been sent from the Guangzhou Center, ready to be sent to the center of Dongguan

* 2018-Aug-29Wed.00:26:08[Dongguan] express mail has arrived in Dongguan center

* 01:30:30[Dongguan] Express has been issued from Dongguan Center, ready to send to Dongguan Shatian

* 04:14:14[Dongguan] Express has arrived in Shatian, Dongguan

* 10:33:37[Dongguan] Dongguan Shatian dispatcher Xiao Baichao 2 18027574003 is sending you a piece

* 15:00:28[Mainland] Your parcel has arrived at the consignment bin.

* 15:01:44[Mainland] Your parcel collection bin has been signed

* 16:37:42[Mainland] Your parcel has been put into storage. If the logistics information has not been updated for a long time, it is recommended to consult the courier company.



“Your parcel has been put into storage” what does this mean? I am in AUSTRALIA not in CHINA, please clarify this for me

Thanks and regards C Bruno

**Comment by bruce356** [comment]
Hi Yongfeng, this is my first purchase from this web site and was not aware that there needed to be a second payment for shipment to Australia. This is the first time I have ever had to do this. Normally shipping is included at time of purchase.

I hope the card will now be sent.

Successfully paid 84.00 yuan!

After the home delivery (single container transportation)parcel arrives at the mainland container warehouse, please operate the consolidating container in time (the maximum free storage for 20 days after the goods arrive in the warehouse)  <http://www.taobao.com/go/chn/transit/index/cht.php> .

 <http://wuliu.taobao.com/user/consolidation/con_transfer_order.htm> View International transport - transport orders  <http://trade.taobao.com/trade/detail/trade_item_detail.htm?biz_order_id=224288327373646697&out_trade_no=T2600P224288327373646697> Order Details

 <http://trip.taobao.com/mytrip/?tvm=tcd&orderId=224288327373646697> Click to receive the mileage : 3 gifts for Ali

Thanks for your Help

Regards – Bruno

**Comment by bobc** [comment]
I would appreciate it if you found somewhere else to communicate instead of spamming my repo. Frankly, I shouldn't have to ask.

Further abuse will result in being blocked.

**Comment by yyf2009** [comment]
@bobc  I am very sorry, please forgive me for my mistake.

**Comment by bob808** [comment]
The board as it stands is not safe. There's a ground problem with this version. I've seen it online on 2.2 and on this 2.3 one that's in this repo. People have made them like this, and I've seen some for sale on various foreign websites.
None of 2.2 and 2.3 are safe, there's a small ground pass of 1mm width that carries current for all mosfets + 3 motor drivers. Anyone with these versions should toss them if they don't understand what needs to be done.
![screen shot 2018-10-14 at 05 55 47](https://user-images.githubusercontent.com/8607897/46912689-36b17680-cf85-11e8-9f53-7cf684197043.png)
![screen shot 2018-10-14 at 07 06 45](https://user-images.githubusercontent.com/8607897/46912692-70827d00-cf85-11e8-9c16-b52288a840f2.png)
